UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -0,0 +1,731 @@
+          PciRootBridgeInfo->RootBridge[index].PMem.Limit           = PcdGet32 (PcdPciReservedPMemLimit);
+          PciRootBridgeInfo->RootBridge[index].UID                  = index;
+          PciRootBridgeInfo->RootBridge[index].HID                  = EISA_PNP_ID (0x0A03);
+
+          Data32 = (UINT32 *)(PropertyPtr->Data);
+          for (Base = 0; Base < 16; Base = Base + 7) {
+            DEBUG ((DEBUG_INFO, "  Base :%x \n", Base));
+            MemType = Fdt32ToCpu (*(Data32 + Base));
+            if (((MemType) & (SS_64BIT_MEMORY_SPACE)) == SS_64BIT_MEMORY_SPACE) {
+              DEBUG ((DEBUG_INFO, "  To program 64 mm \n"));
+              PciRootBridgeInfo->RootBridge[index].MemAbove4G.Base  = Fdt32ToCpu (*(Data32 + Base + 2)) + LShiftU64 (Fdt32ToCpu (*(Data32 + Base + 1)), 32);
+              PciRootBridgeInfo->RootBridge[index].MemAbove4G.Limit = PciRootBridgeInfo->RootBridge[index].MemAbove4G.Base + LShiftU64 (Fdt32ToCpu (*(Data32 + Base + 5)), 32) +  Fdt32ToCpu (*(Data32 + Base + 6)) -1;
+            } else if (((MemType) & (SS_32BIT_MEMORY_SPACE)) == SS_32BIT_MEMORY_SPACE) {
+              DEBUG ((DEBUG_INFO, "  To program 32 mem \n"));
+              PciRootBridgeInfo->RootBridge[index].Mem.Base  = Fdt32ToCpu (*(Data32 + Base + 2));
+              PciRootBridgeInfo->RootBridge[index].Mem.Limit = PciRootBridgeInfo->RootBridge[index].Mem.Base + Fdt32ToCpu (*(Data32 + Base + 6)) -1;
+            } else if (((MemType) & (SS_IO_SPACE)) == SS_IO_SPACE) {
+              DEBUG ((DEBUG_INFO, "  To program Io\n"));
+              PciRootBridgeInfo->RootBridge[index].Io.Base  = Fdt32ToCpu (*(Data32 + Base + 2));
+              PciRootBridgeInfo->RootBridge[index].Io.Limit = PciRootBridgeInfo->RootBridge[index].Io.Base + Fdt32ToCpu (*(Data32 +  Base + 6)) -1;
+            }
+          }
+
+          DEBUG ((DEBUG_INFO, "RootBridgeCount %x, index :%x\n", RootBridgeCount, index));
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Mem.Base %x, \n", PciRootBridgeInfo->RootBridge[index].Mem.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Mem.limit %x, \n", PciRootBridgeInfo->RootBridge[index].Mem.Limit));
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->MemAbove4G.Base %llx, \n", PciRootBridgeInfo->RootBridge[index].MemAbove4G.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->MemAbove4G.limit %llx, \n", PciRootBridgeInfo->RootBridge[index].MemAbove4G.Limit));
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Io.Base %llx, \n", PciRootBridgeInfo->RootBridge[index].Io.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Io.limit %llx, \n", PciRootBridgeInfo->RootBridge[index].Io.Limit));
+        }
+
+        if (AsciiStrCmp (TempStr, "bus-range") == 0) {
+          DEBUG ((DEBUG_INFO, "  Found bus-range Property TempLen (%08X)\n", TempLen));
+
+          Data16                                               = (UINT16 *)(PropertyPtr->Data);

```

Comment:
As per DT specification table 2.3 shouldn't the size be u32 or u64 for the node?

Reply:
With that pci-enum-done node should also change.

Reply:
yes, for bus-range we will use u32 type as it actually only u8 data valid. later if we need to support both u32 and u64 we may create PCD or MACRO to replace it.

Reply:
resolved